packages/backend/src/utils/recipeStatusUtils.ts

```diff
@@ -46,7 +46,22 @@ export class RecipeStatusUtils {
     this.update();
   }
 
-  setTaskState(taskId: string, state: TaskState) {
+  setTaskError(taskId: string, error: string) {
+    if (!this.tasks.has(taskId)) throw new Error('task not found.');
+    const task = this.tasks.get(taskId);
+    this.setTask({
+      // we place the error before, as we do not want to overwrite any existing error (we try to keep the first one)
```

Comment:
Feel free to add some unit tests to guarantee this

Reply:
Yeah i'm writing them !
> I should have set it as draft sorry